### PR TITLE
[dy] View source preview

### PR DESCRIPTION
**Pull request by dy46**
# Summary

View data preview for data integration sources. It currently doesn't work for all sources, but it should fail gracefully (there just won't be a data preview) for unsupported sources. I would like to merge this in to avoid conflicts for some of the other integration pipeline changes.

# Tests

tested locally

Screenshot 2022-11-09 at 4 48 35 PM

cc: @wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/sources/sql/base.py [hunk 1]
```diff
@@ -28,11 +29,13 @@ class Source(BaseSource):
     def table_prefix(self):
         return ''
 
+    @abstractmethod
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/sql/base.py:32]: To use `abstractmethod`, the class needs to inherit from `ABC`

File: mage_integrations/mage_integrations/sources/postgresql/__init__.py [hunk 2]
```diff
@@ -46,6 +46,10 @@ def build_discover_query(self, streams: List[str] = None):
             query = f'{query}\nAND c.TABLE_NAME IN ({table_names})'
         return query
 
+    def update_column_names(self, columns: List[str]) -> List[str]:
+        return list(map(lambda column: f'"{column}"', columns))
+
+
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/postgresql/__init__.py:52]: nit: remove extra blank lines

File: mage_integrations/mage_integrations/sources/base.py [hunk 3]
```diff
@@ -178,6 +191,25 @@ def process(self) -> None:
         try:
             if self.should_test_connection:
                 self.test_connection()
+            elif self.load_sample_data:
+                catalog = self.catalog or self.discover(streams=self.selected_streams)
+                for stream in catalog.get_selected_streams(self.state):
+                    gen = self.load_data(stream, sample_data=True)
+                    if gen is not None \
+                        and self.pipeline_uuid is not None \
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/base.py:199]: ideally, we shouldn't use pipeline_uuid and block_uuid here. mage_integrations module just output the data. then mage_ai can parse the output and store the dataframe.

File: mage_integrations/mage_integrations/sources/base.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from datetime import datetime
+from mage_ai.data_preparation.models.pipeline import Pipeline
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/base.py:2]: I mentioned this issue before. We shouldn't import `mage_ai` here.

**Comment by dy46** [comment]
@wangxiaoyou1993 updated

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**